docs/implementation/user_api/Authentication.md

```diff
@@ -0,0 +1,106 @@
+
+While the server is waiting for proof of possession, the client is polling the server to determine if the user has achieved authentication. This behaviour is **inspired** by the [OpenID Connect CIBA Poll Mode](https://openid.net/specs/openid-client-initiated-backchannel-authentication-core-1_0.html#rfc.section.5). In the future the Authentication Flow may support notifications through WebSockets to avoid polling the server.
+
+### Authentication Methods
+
+For the email authentication method the flow #2 should return:
+
+```json
+[
+    {
+        "type": "email",
+        "allowed_domains": ["*.isel.pt", "*.isel.ipl.pt"]
+    },
+]
+```
+
+Where, in this case, the `allowed_domains` allows emails from the specified domains:
+
+- Allows `isel.pt` and `isel.ipl.pt`
+- Allows subdomains of `isel.pt` and `isel.ipl.pt` such as `alunos.isel.pt`
+
+In the flow #6 the client must send the authentication request to the server. Below an example of this request is presented for the email authentication method.
+
+```json
+{
+    "scope": "profile",
+    "type": "email",
+    "client_id": "The Client ID",
+    "notification_method": "The method used to notify the client about the authentication response, for now only 'POLL' is supported",
+    "login_hint": "AXXXXX@alunos.isel.pt"
+}
+```
+
+For the `POLL` notification method the response is for example:
+
+```json
+{
+    "auth_req_id": "Unique ID of the Authentication Request",
+    "expires_in": 120,
+    "interval": 2
+}
+```
+
+This response follows the [Successful Authentication Request Acknowledgement](https://openid.net/specs/openid-client-initiated-backchannel-authentication-core-1_0.html#successful_authentication_request_acknowdlegment) of the OpenID Client Initiated Backchannel Authentication.
+
+If an error has occurred the [Authentication Error Response](https://openid.net/specs/openid-client-initiated-backchannel-authentication-core-1_0.html#auth_error_response) applies.
+
+After the flow #6 the server will send an email challenge to the user email (for example, `AXXXXX@alunos.isel.pt`), whilst the user needs to acknowledge this challenge by logging in to the `alunos.isel.pt` email server and accepting the challenge (a URL that redirects back to the server).
+
+If the `POLL` method was chosen the user should poll the server, with the specified interval between poll requests, and the server will respond if the authentication request has been fulfilled by the user.
+
+The responses sent by the server to the client if an error has been found, or if the user hasn't fulfilled the request follow the [OpenID Token Error Response](https://openid.net/specs/openid-connect-core-1_0.html#TokenErrorResponse).
+
+If the user has fulfilled the authorization request the response is presented as follows:
+
+```json
```

Comment:
Do we need the `code` intermediate step? IINM, CIBA doesn't require. The AS sends the token response (including the `access_token` directly in the response.